google/cloud/monitoring_v3/query.py

```diff
@@ -460,12 +461,16 @@ def _build_query_params(self, headers_only=False, page_size=None):
             from this request. Non-positive values are ignored. Defaults
             to a sensible value set by the API.
         """
-        params = {"name": self._project_path, "filter_": self.filter}
+        params = {"name": self._project_path, "filter": self.filter}
 
+        end_timestamp = timestamp.Timestamp()
+        end_timestamp.FromDatetime(self._end_time)
         params["interval"] = types.TimeInterval()
-        params["interval"].end_time.FromDatetime(self._end_time)
+        params["interval"].end_time = end_timestamp
         if self._start_time:
```

Comment:
We _may_ not need the conditional. If `self._start_time` is `None` there shouldn't be a problem setting it.

If this is true, we can simplify the interval construction a *lot*.
```python
params = {..., "interval": types.TimeInterval(start_time=self._start_time, end_time=self._end_time)}
```

Reply:
done